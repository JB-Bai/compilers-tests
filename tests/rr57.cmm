int SJbhqVf(){
	int Z5D3bQ8 =(i9zlZSIu5()) && 0x28 + 41226 * XK6ZIQMHna * 527.0E+2, VmP6N = 1817.f_gazm, cKoNMzcaD[0435] = 0266 >(956.3802E9)[- 746943 && 217.1487 || 7717[16.9E-2] == 2516 * 48577 == f_dIh03nQP[841.98e-8] && 03.6450] > JRFuYE[981.68E2], Mm5pa = Q3FQ0f6();
	struct UkH0R8 kVadc9tCQ[0x9] = - 0x067e, yyPpn3k = 071567.KOkQcoL * SZrAh(FnXdLiNuW, 0x4e, 060, -(06.25e7 + 70.274 * 7857.17 / 00352 * bDBC9dv = 8.51 || ! ZFproOh51) = 05455 && GQClT + sbPOaE_T, obCHRW, - 047, 19.8847e3[WHqpFu || Yaj6DpnC0 || KYEMt4 / 7.96E+9 * 0x6456.aaEjtGpS.u7PCaKjX]) / ! HHq9o_v - 419;
	int GVAf[6946][0x87da] =(459);
	if(aiQkYiO_c[(paPw3BtZ * 015 / 380.19 + 046316)] - Mzxa2TIB) 282444;
}
struct bykOM0W ROFU9w(){
	struct FjCZgbe8tf Cc4x7wPV, hQmZ1wZV = gza9s5, hVqSlU = 3551.704 + - UWVWFHM9, Xz0QRDXAKh = 00204;
	float c04m_k0 = 901.7588e-7;
	{
		gbo42Xb.B_RnT;
		l0yHV4w - reYLRBDZ9gG;
	}
}
int KnaAV, BZkIdR;
float BMXxgUbP(){
	float RDxc3FmBI = wqzi4cfR + IOyTNji && XsWaxOA7(7920 = 31.109) * 0xaBB[2830.966e8] = - 0xb01 > MS8_(062636) || 634.0477[- 362.78E+8 * IB7yzyIH() * 25.92 && jO1kfMe <= ! 384.45 || 0x0C7b - xuFC2Wwz2z().gmtWnCjjY + 05727.uoFYsC && !(09.729E6) - 761.15E+9] = sChFPelIG + sjjR32iI().s0HvVJi / 827.453e4 && - jjhdLfF(0x1, Ttf81wGu() &&(- - W5kcwgI1W), 047022, O7DLq3rYF, 60.8e0, -(680.699e3) + 0x4) - 849.9611 != TIT0uOP - P_4TLZucBb < MtrlB, xfCWjsk9, svrM40RJ[0575] = 05475[090.08], kS8Wgk;
	float VzyOR9y1, J_kw_lN = 39.849e-4 && - 0xcB97;
	_BOXxxQ || 0x97 || 06.2474 >= YEU6DHi;
	627 + 6.530[- uo0Rxm2s];
	UQOcAmW8 && 05375 * 62387 + 49.0866 <= owNVemf - 4208.83 +(79.431E4.VqViBmGr);
	- 9768.78;
}

