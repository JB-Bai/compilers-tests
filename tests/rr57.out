Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: SJbhqVf
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: Z5D3bQ8
                ASSIGNOP
                Exp (2)
                  Exp (2)
                    LP
                    Exp (2)
                      ID: i9zlZSIu5
                      LP
                      RP
                    RP
                  AND
                  Exp (2)
                    Exp (2)
                      INT: 40
                    PLUS
                    Exp (2)
                      Exp (2)
                        Exp (2)
                          INT: 41226
                        STAR
                        Exp (2)
                          ID: XK6ZIQMHna
                      STAR
                      Exp (2)
                        FLOAT: 52700.000000
              COMMA
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: VmP6N
                  ASSIGNOP
                  Exp (2)
                    Exp (2)
                      INT: 1817
                    DOT
                    ID: f_gazm
                COMMA
                DecList (2)
                  Dec (2)
                    VarDec (2)
                      VarDec (2)
                        ID: cKoNMzcaD
                      LB
                      INT: 285
                      RB
                    ASSIGNOP
                    Exp (2)
                      Exp (2)
                        Exp (2)
                          INT: 182
                        RELOP
                        Exp (2)
                          Exp (2)
                            LP
                            Exp (2)
                              FLOAT: 956380217344.000000
                            RP
                          LB
                          Exp (2)
                            Exp (2)
                              Exp (2)
                                MINUS
                                Exp (2)
                                  INT: 746943
                              AND
                              Exp (2)
                                FLOAT: 217.148697
                            OR
                            Exp (2)
                              Exp (2)
                                Exp (2)
                                  Exp (2)
                                    Exp (2)
                                      INT: 7717
                                    LB
                                    Exp (2)
                                      FLOAT: 0.169000
                                    RB
                                  RELOP
                                  Exp (2)
                                    Exp (2)
                                      INT: 2516
                                    STAR
                                    Exp (2)
                                      INT: 48577
                                RELOP
                                Exp (2)
                                  Exp (2)
                                    ID: f_dIh03nQP
                                  LB
                                  Exp (2)
                                    FLOAT: 0.000008
                                  RB
                              AND
                              Exp (2)
                                FLOAT: 3.645000
                          RB
                      RELOP
                      Exp (2)
                        Exp (2)
                          ID: JRFuYE
                        LB
                        Exp (2)
                          FLOAT: 98168.000000
                        RB
                  COMMA
                  DecList (2)
                    Dec (2)
                      VarDec (2)
                        ID: Mm5pa
                      ASSIGNOP
                      Exp (2)
                        ID: Q3FQ0f6
                        LP
                        RP
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                StructSpecifier (3)
                  STRUCT
                  Tag (3)
                    ID: UkH0R8
              DecList (3)
                Dec (3)
                  VarDec (3)
                    VarDec (3)
                      ID: kVadc9tCQ
                    LB
                    INT: 9
                    RB
                  ASSIGNOP
                  Exp (3)
                    MINUS
                    Exp (3)
                      INT: 1662
                COMMA
                DecList (3)
                  Dec (3)
                    VarDec (3)
                      ID: yyPpn3k
                    ASSIGNOP
                    Exp (3)
                      Exp (3)
                        Exp (3)
                          Exp (3)
                            Exp (3)
                              INT: 29559
                            DOT
                            ID: KOkQcoL
                          STAR
                          Exp (3)
                            ID: SZrAh
                            LP
                            Args (3)
                              Exp (3)
                                ID: FnXdLiNuW
                              COMMA
                              Args (3)
                                Exp (3)
                                  INT: 78
                                COMMA
                                Args (3)
                                  Exp (3)
                                    INT: 48
                                  COMMA
                                  Args (3)
                                    Exp (3)
                                      Exp (3)
                                        MINUS
                                        Exp (3)
                                          LP
                                          Exp (3)
                                            Exp (3)
                                              Exp (3)
                                                FLOAT: 62500000.000000
                                              PLUS
                                              Exp (3)
                                                Exp (3)
                                                  Exp (3)
                                                    Exp (3)
                                                      FLOAT: 70.274002
                                                    STAR
                                                    Exp (3)
                                                      FLOAT: 7857.169922
                                                  DIV
                                                  Exp (3)
                                                    INT: 234
                                                STAR
                                                Exp (3)
                                                  ID: bDBC9dv
                                            ASSIGNOP
                                            Exp (3)
                                              Exp (3)
                                                FLOAT: 8.510000
                                              OR
                                              Exp (3)
                                                NOT
                                                Exp (3)
                                                  ID: ZFproOh51
                                          RP
                                      ASSIGNOP
                                      Exp (3)
                                        Exp (3)
                                          INT: 2861
                                        AND
                                        Exp (3)
                                          Exp (3)
                                            ID: GQClT
                                          PLUS
                                          Exp (3)
                                            ID: sbPOaE_T
                                    COMMA
                                    Args (3)
                                      Exp (3)
                                        ID: obCHRW
                                      COMMA
                                      Args (3)
                                        Exp (3)
                                          MINUS
                                          Exp (3)
                                            INT: 39
                                        COMMA
                                        Args (3)
                                          Exp (3)
                                            Exp (3)
                                              FLOAT: 19884.699219
                                            LB
                                            Exp (3)
                                              Exp (3)
                                                Exp (3)
                                                  ID: WHqpFu
                                                OR
                                                Exp (3)
                                                  ID: Yaj6DpnC0
                                              OR
                                              Exp (3)
                                                Exp (3)
                                                  Exp (3)
                                                    ID: KYEMt4
                                                  DIV
                                                  Exp (3)
                                                    FLOAT: 7960000000.000000
                                                STAR
                                                Exp (3)
                                                  Exp (3)
                                                    Exp (3)
                                                      INT: 25686
                                                    DOT
                                                    ID: aaEjtGpS
                                                  DOT
                                                  ID: u7PCaKjX
                                            RB
                            RP
                        DIV
                        Exp (3)
                          NOT
                          Exp (3)
                            ID: HHq9o_v
                      MINUS
                      Exp (3)
                        INT: 419
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: int
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      VarDec (4)
                        VarDec (4)
                          ID: GVAf
                        LB
                        INT: 6946
                        RB
                      LB
                      INT: 34778
                      RB
                    ASSIGNOP
                    Exp (4)
                      LP
                      Exp (4)
                        INT: 459
                      RP
                SEMI
        StmtList (5)
          Stmt (5)
            IF
            LP
            Exp (5)
              Exp (5)
                Exp (5)
                  ID: aiQkYiO_c
                LB
                Exp (5)
                  LP
                  Exp (5)
                    Exp (5)
                      Exp (5)
                        Exp (5)
                          ID: paPw3BtZ
                        STAR
                        Exp (5)
                          INT: 13
                      DIV
                      Exp (5)
                        FLOAT: 380.190002
                    PLUS
                    Exp (5)
                      INT: 19662
                  RP
                RB
              MINUS
              Exp (5)
                ID: Mzxa2TIB
            RP
            Stmt (5)
              Exp (5)
                INT: 282444
              SEMI
        RC
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          StructSpecifier (7)
            STRUCT
            Tag (7)
              ID: bykOM0W
        FunDec (7)
          ID: ROFU9w
          LP
          RP
        CompSt (7)
          LC
          DefList (8)
            Def (8)
              Specifier (8)
                StructSpecifier (8)
                  STRUCT
                  Tag (8)
                    ID: FjCZgbe8tf
              DecList (8)
                Dec (8)
                  VarDec (8)
                    ID: Cc4x7wPV
                COMMA
                DecList (8)
                  Dec (8)
                    VarDec (8)
                      ID: hQmZ1wZV
                    ASSIGNOP
                    Exp (8)
                      ID: gza9s5
                  COMMA
                  DecList (8)
                    Dec (8)
                      VarDec (8)
                        ID: hVqSlU
                      ASSIGNOP
                      Exp (8)
                        Exp (8)
                          FLOAT: 3551.704102
                        PLUS
                        Exp (8)
                          MINUS
                          Exp (8)
                            ID: UWVWFHM9
                    COMMA
                    DecList (8)
                      Dec (8)
                        VarDec (8)
                          ID: Xz0QRDXAKh
                        ASSIGNOP
                        Exp (8)
                          INT: 132
              SEMI
            DefList (9)
              Def (9)
                Specifier (9)
                  TYPE: float
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: c04m_k0
                    ASSIGNOP
                    Exp (9)
                      FLOAT: 0.000090
                SEMI
          StmtList (10)
            Stmt (10)
              CompSt (10)
                LC
                StmtList (11)
                  Stmt (11)
                    Exp (11)
                      Exp (11)
                        ID: gbo42Xb
                      DOT
                      ID: B_RnT
                    SEMI
                  StmtList (12)
                    Stmt (12)
                      Exp (12)
                        Exp (12)
                          ID: l0yHV4w
                        MINUS
                        Exp (12)
                          ID: reYLRBDZ9gG
                      SEMI
                RC
          RC
      ExtDefList (15)
        ExtDef (15)
          Specifier (15)
            TYPE: int
          ExtDecList (15)
            VarDec (15)
              ID: KnaAV
            COMMA
            ExtDecList (15)
              VarDec (15)
                ID: BZkIdR
          SEMI
        ExtDefList (16)
          ExtDef (16)
            Specifier (16)
              TYPE: float
            FunDec (16)
              ID: BMXxgUbP
              LP
              RP
            CompSt (16)
              LC
              DefList (17)
                Def (17)
                  Specifier (17)
                    TYPE: float
                  DecList (17)
                    Dec (17)
                      VarDec (17)
                        ID: RDxc3FmBI
                      ASSIGNOP
                      Exp (17)
                        Exp (17)
                          Exp (17)
                            Exp (17)
                              ID: wqzi4cfR
                            PLUS
                            Exp (17)
                              ID: IOyTNji
                          AND
                          Exp (17)
                            Exp (17)
                              ID: XsWaxOA7
                              LP
                              Args (17)
                                Exp (17)
                                  Exp (17)
                                    INT: 7920
                                  ASSIGNOP
                                  Exp (17)
                                    FLOAT: 31.108999
                              RP
                            STAR
                            Exp (17)
                              Exp (17)
                                INT: 2747
                              LB
                              Exp (17)
                                FLOAT: 283096612864.000000
                              RB
                        ASSIGNOP
                        Exp (17)
                          Exp (17)
                            Exp (17)
                              Exp (17)
                                MINUS
                                Exp (17)
                                  INT: 2817
                              RELOP
                              Exp (17)
                                ID: MS8_
                                LP
                                Args (17)
                                  Exp (17)
                                    INT: 26014
                                RP
                            OR
                            Exp (17)
                              Exp (17)
                                FLOAT: 634.047729
                              LB
                              Exp (17)
                                Exp (17)
                                  Exp (17)
                                    Exp (17)
                                      Exp (17)
                                        MINUS
                                        Exp (17)
                                          FLOAT: 36278001664.000000
                                      STAR
                                      Exp (17)
                                        ID: IB7yzyIH
                                        LP
                                        RP
                                    STAR
                                    Exp (17)
                                      FLOAT: 25.920000
                                  AND
                                  Exp (17)
                                    Exp (17)
                                      ID: jO1kfMe
                                    RELOP
                                    Exp (17)
                                      NOT
                                      Exp (17)
                                        FLOAT: 384.450012
                                OR
                                Exp (17)
                                  Exp (17)
                                    Exp (17)
                                      Exp (17)
                                        INT: 3195
                                      MINUS
                                      Exp (17)
                                        Exp (17)
                                          ID: xuFC2Wwz2z
                                          LP
                                          RP
                                        DOT
                                        ID: gmtWnCjjY
                                    PLUS
                                    Exp (17)
                                      Exp (17)
                                        INT: 3031
                                      DOT
                                      ID: uoFYsC
                                  AND
                                  Exp (17)
                                    Exp (17)
                                      NOT
                                      Exp (17)
                                        LP
                                        Exp (17)
                                          FLOAT: 9729000.000000
                                        RP
                                    MINUS
                                    Exp (17)
                                      FLOAT: 761149980672.000000
                              RB
                          ASSIGNOP
                          Exp (17)
                            Exp (17)
                              Exp (17)
                                ID: sChFPelIG
                              PLUS
                              Exp (17)
                                Exp (17)
                                  Exp (17)
                                    ID: sjjR32iI
                                    LP
                                    RP
                                  DOT
                                  ID: s0HvVJi
                                DIV
                                Exp (17)
                                  FLOAT: 8274530.000000
                            AND
                            Exp (17)
                              Exp (17)
                                Exp (17)
                                  Exp (17)
                                    MINUS
                                    Exp (17)
                                      ID: jjhdLfF
                                      LP
                                      Args (17)
                                        Exp (17)
                                          INT: 1
                                        COMMA
                                        Args (17)
                                          Exp (17)
                                            Exp (17)
                                              ID: Ttf81wGu
                                              LP
                                              RP
                                            AND
                                            Exp (17)
                                              LP
                                              Exp (17)
                                                MINUS
                                                Exp (17)
                                                  MINUS
                                                  Exp (17)
                                                    ID: W5kcwgI1W
                                              RP
                                          COMMA
                                          Args (17)
                                            Exp (17)
                                              INT: 19986
                                            COMMA
                                            Args (17)
                                              Exp (17)
                                                ID: O7DLq3rYF
                                              COMMA
                                              Args (17)
                                                Exp (17)
                                                  FLOAT: 60.799999
                                                COMMA
                                                Args (17)
                                                  Exp (17)
                                                    Exp (17)
                                                      MINUS
                                                      Exp (17)
                                                        LP
                                                        Exp (17)
                                                          FLOAT: 680699.000000
                                                        RP
                                                    PLUS
                                                    Exp (17)
                                                      INT: 4
                                      RP
                                  MINUS
                                  Exp (17)
                                    FLOAT: 849.961121
                                RELOP
                                Exp (17)
                                  Exp (17)
                                    ID: TIT0uOP
                                  MINUS
                                  Exp (17)
                                    ID: P_4TLZucBb
                              RELOP
                              Exp (17)
                                ID: MtrlB
                    COMMA
                    DecList (17)
                      Dec (17)
                        VarDec (17)
                          ID: xfCWjsk9
                      COMMA
                      DecList (17)
                        Dec (17)
                          VarDec (17)
                            VarDec (17)
                              ID: svrM40RJ
                            LB
                            INT: 381
                            RB
                          ASSIGNOP
                          Exp (17)
                            Exp (17)
                              INT: 2877
                            LB
                            Exp (17)
                              FLOAT: 90.080002
                            RB
                        COMMA
                        DecList (17)
                          Dec (17)
                            VarDec (17)
                              ID: kS8Wgk
                  SEMI
                DefList (18)
                  Def (18)
                    Specifier (18)
                      TYPE: float
                    DecList (18)
                      Dec (18)
                        VarDec (18)
                          ID: VzyOR9y1
                      COMMA
                      DecList (18)
                        Dec (18)
                          VarDec (18)
                            ID: J_kw_lN
                          ASSIGNOP
                          Exp (18)
                            Exp (18)
                              FLOAT: 0.003985
                            AND
                            Exp (18)
                              MINUS
                              Exp (18)
                                INT: 52119
                    SEMI
              StmtList (19)
                Stmt (19)
                  Exp (19)
                    Exp (19)
                      Exp (19)
                        ID: _BOXxxQ
                      OR
                      Exp (19)
                        INT: 151
                    OR
                    Exp (19)
                      Exp (19)
                        FLOAT: 6.247400
                      RELOP
                      Exp (19)
                        ID: YEU6DHi
                  SEMI
                StmtList (20)
                  Stmt (20)
                    Exp (20)
                      Exp (20)
                        INT: 627
                      PLUS
                      Exp (20)
                        Exp (20)
                          FLOAT: 6.530000
                        LB
                        Exp (20)
                          MINUS
                          Exp (20)
                            ID: uo0Rxm2s
                        RB
                    SEMI
                  StmtList (21)
                    Stmt (21)
                      Exp (21)
                        Exp (21)
                          ID: UQOcAmW8
                        AND
                        Exp (21)
                          Exp (21)
                            Exp (21)
                              Exp (21)
                                INT: 2813
                              STAR
                              Exp (21)
                                INT: 62387
                            PLUS
                            Exp (21)
                              FLOAT: 49.086601
                          RELOP
                          Exp (21)
                            Exp (21)
                              Exp (21)
                                ID: owNVemf
                              MINUS
                              Exp (21)
                                FLOAT: 4208.830078
                            PLUS
                            Exp (21)
                              LP
                              Exp (21)
                                Exp (21)
                                  FLOAT: 794310.000000
                                DOT
                                ID: VqViBmGr
                              RP
                      SEMI
                    StmtList (22)
                      Stmt (22)
                        Exp (22)
                          MINUS
                          Exp (22)
                            FLOAT: 9768.780273
                        SEMI
              RC
